### PR TITLE
Added SubitemMetadata struct to share with Packager

### PR DESCRIPTION
**Pull request by stephanheilner**
* I added some additional info for a unique constraint error we're seeing when adding nav collections. We can't see what uri is causing it, and we've had a really hard time tracking down how and why its failing. This should provide a little extra data, but still throw the exception correctly.

### CODE REVIEW

**Comment by stephanheilner** [comment]
Tests pass locally.

File: LDSContent/SubitemMetadata.swift [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+        self.docID = docID
+        self.docVersion = docVersion
+    }
+    
+    public var hashValue: Int {
+        return "\(itemID),\(subitemID),\(docID),\(docVersion)".hashValue
+    }
+}
+
+public func == (lhs: SubitemMetadata, rhs: SubitemMetadata) -> Bool {
```
Comment (nickmshelley) [inline LDSContent/SubitemMetadata.swift]: If you're already implementing `==` you might as well make the struct `Equatable`.

File: LDSContent/SubitemMetadata.swift [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    
+    public init(itemID: Int64, subitemID: Int64, docID: String, docVersion: Int) {
+        self.itemID = itemID
+        self.subitemID = subitemID
+        self.docID = docID
+        self.docVersion = docVersion
+    }
+    
+    public var hashValue: Int {
+        return "\(itemID),\(subitemID),\(docID),\(docVersion)".hashValue
```
Comment (nickmshelley) [inline LDSContent/SubitemMetadata.swift]: Not important, but you probably don't need the commas to get the hash.

File: LDSContent/SubitemMetadata.swift (hunk 1 above)
Comment (nickmshelley) [inline LDSContent/SubitemMetadata.swift]: @stephanheilner pointed out that `public protocol Hashable : Equatable`

**Comment by stephanheilner** [comment]
Tests still pass locally.

File: LDSContent/SubitemMetadata.swift (hunk 1 above)
Comment (hiltonc) [inline LDSContent/SubitemMetadata.swift]: Super minor, but now that Swift allows == to be a static method, I prefer to put it inside the braces, like:

```swift
    ...

    public var hashValue: Int { ... }

    public static func == (lhs: SubitemMetadata, rhs: SubitemMetadata) -> Bool { ... }
}
```

**Pull request merged**